plugins/modules/s3_object.py

```diff
@@ -654,15 +654,9 @@ def path_check(path):
         return False
 
 
-def get_content_type(src, present=True):
-    if not present:
-        content_type = None
-        if src:
-            content_type = mimetypes.guess_type(src)[0]
-        if content_type is None:
-            # s3 default content type
-            content_type = "binary/octet-stream"
-        return content_type
+def guess_content_type(src):
+    content_type = mimetypes.guess_type(src)[0] if src else None
+    return content_type if content_type else "binary/octet-stream"
```

Comment:
Thinking about it a little, the None assignment isn't actually needed:

```suggestion
    if src:
        content_type = mimetypes.guess_type(src)[0]
        if content_type:
            return content_type

    # S3 default content type
    return "binary/octet-stream"
```

(btw +1 to renaming the function)